From a70e7cd62385ca325b394926096ce7c93f57c3c0 Mon Sep 17 00:00:00 2001
From: Youling Tang <tangyouling@kylinos.cn>
Date: Mon, 30 Mar 2026 03:53:37 +0000
Subject: [PATCH] smb/client: check if mapping_table_ERRDOS and
 mapping_table_ERRSRV are sorted

Although the arrays are sorted at build time, verify the ordering again
when cifs.ko is loaded to avoid potential regressions introduced by
future script changes.

Signed-off-by: Youling Tang <tangyouling@kylinos.cn>
Reviewed-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/client/smb1maperror.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/fs/smb/client/smb1maperror.c b/fs/smb/client/smb1maperror.c
index be88a949c72d..bed20b0cc9b5 100644
--- a/fs/smb/client/smb1maperror.c
+++ b/fs/smb/client/smb1maperror.c
@@ -213,6 +213,10 @@ static int __array ## _check_sort(void)					\
 
 /* ntstatus_to_dos_map_check_sort */
 DEFINE_CHECK_SORT_FUNC(ntstatus_to_dos_map, ntstatus);
+/* mapping_table_ERRDOS_check_sort */
+DEFINE_CHECK_SORT_FUNC(mapping_table_ERRDOS, smb_err);
+/* mapping_table_ERRSRV_check_sort */
+DEFINE_CHECK_SORT_FUNC(mapping_table_ERRSRV, smb_err);
 
 int __init smb1_init_maperror(void)
 {
@@ -222,5 +226,13 @@ int __init smb1_init_maperror(void)
 	if (rc)
 		return rc;
 
+	rc = mapping_table_ERRDOS_check_sort();
+	if (rc)
+		return rc;
+
+	rc = mapping_table_ERRSRV_check_sort();
+	if (rc)
+		return rc;
+
 	return rc;
 }
-- 
2.52.0

